app/controllers/courses_controller.rb

```diff
@@ -215,6 +216,10 @@ def set_endpoint
     @endpoint = params[:endpoint] if SHOW_ENDPOINTS.include?(params[:endpoint])
   end
 
+  def set_limit
+    @limit = (params[:limit] || 50)
```

Comment:
This points to a problem with how we've set up the multipurpose `show` endpoint for courses. The limit is relevant only for a single endpoint, but this will set it for every endpoint. In the short term, another option might be to handle the limit param inside the jbuilder view.

In the longer term, I think we need to rewrite the routing and controller to have an explicit endpoint for each type of json data for a course.

Reply:
I limited the set_limit to the revisions endpoint with an IF statement, as I think it's clearer if it's in the controller. Why do you think it's better to add it to the json jbuilder?

Reply:
I'm okay with the if statement. That should make it clear that this is only relevant for the one endpoint.